spec.bs

```diff
@@ -232,7 +232,105 @@ to conduct an auction to select an advertisement to display to the user, they ca
 {{Window/navigator}}.{{Navigator/runAdAuction()}} function, providing an auction configuration that
 tells the browser how to conduct the auction and which on-device recorded interests are allowed to
 bid in the auction for the chance to display their advertisement.
-  
+
+<h3 id="script-runners">Script Runners</h3>
+
+<xmp class="idl">
+[Exposed=Window, Global=Window]
+interface InterestGroupBiddingScriptRunnerGlobalScope{
+};
+
+[Exposed=Window, Global=Window]
+interface InterestGroupScoringScriptRunnerGlobalScope{
+};
+
+[Exposed=Window, Global=Window]
+interface InterestGroupReportingScriptRunnerGlobalScope{
+};
+</xmp>
+
+Issue: where do these interfaces above get exposed to?
+
+TODO: modelingSignals
+
+<xmp class="idl">
+[Exposed=InterestGroupBiddingScriptRunnerGlobalScope]
+dictionary AdSize {
+  required double width;
+  required DOMString widthUnits;
+  required double height;
+  required DOMString heightUnits;
+};
+
+[Exposed=InterestGroupBiddingScriptRunnerGlobalScope]
+dictionary AdDescriptor {
+  required DOMString render;
+  AdSize adSize;
+};
+
+[Exposed=InterestGroupBiddingScriptRunnerGlobalScope]
+dictionary GenerateBidOutput {
+  required double bid;
+  required AdDescriptor adDescriptor;
+  DOMString ad;
+  double adCost;
+  sequence<AdDescriptor> adComponentDescriptors;
+  boolean allowComponentAuction = false;
+};
+
+</xmp>
+
+TODO: define an enum LengthUnit for AdSize's widthUnits and heightUnits.
+
+<xmp class="idl">
+[Exposed=(InterestGroupBiddingScriptRunnerGlobalScope, InterestGroupScoringScriptRunnerGlobalScope, 

```

Comment:
If we adopt my above suggestion, then we can make this `Exposed=InterestGroupScriptRunner` without explicitly enumerating each global type.